Use "nested flow" in place of "subflow" in the docs (#14122)
PrefectHQ/prefect: docs/2.19.x/cloud/api-rate-limits.mdx

Note that, because Flow Run Retention Policy keys on terminal state, if two flows start at the same time, but reach a terminal state at different times, they will be removed at different times according to when they each reached their respective terminal states.
</Tip>

> This retention policy applies to all [details about a flow run](https://docs.prefect.io/ui/flow-runs/#inspect-a-flow-run), including its task runs. Nested flow runs follow the retention policy independently from their parent flow runs, and are removed based on the time each nested flow run reaches a terminal state.

Reviewer: ```suggestion
This retention policy applies to all [details about a flow run](https://docs.prefect.io/ui/flow-runs/#inspect-a-flow-run), including its task runs. 
Nested flow runs are retained independently from their parent flow runs. They are deleted based on the time each nested flow run reaches a terminal state.
```

Author: ```suggestion
This retention policy applies to all [details about a flow run](https://docs.prefect.io/ui/flow-runs/#inspect-a-flow-run), including its task runs. 
Nested flow runs are retained independently from their parent flow runs. Nested flow runs are deleted based on the time they reached a terminal state.
```

Author: Thank you. Updated further @billpalombi
